### DISCUSSION TITLE
Struggeling with power calculation of my Altherma 3 R Ech2O (1phase)

### DISCUSSION THREAD

**Discussion by nicx** [raomin/ESPAltherma#354] [General]
Hi,

im am struggling with the power calculation of my Altherma 3 R Ech2O (1phase 8kW Model). It's working somehow, but I cannot believe the values. 

This is what I get via ESPAltherma:
INV Primärstrom (A) 3,5
Stromaufnahme INV-Verdichter (A) 8,5
Spannung (V) 75

When calculating the power I will get 3,5 * 75 = 262,5 W.
But when looking on my main power smart meter the power of the heat pump should be more or less around 600-800W.

First Question: Why are there different current values reported (3,5A and 8,5A)?
Second Question: Why is ESPAltherma reporting 75V? Is this right or should I use 230V for the power calculation?

Thanks for your help! :)

**Comment by KalaNagHTD** [comment]
You have the wrong voltage!
Best is calculating with 230V.

**Reply by nicx** [reply]
@KalaNagHTD ok, thanks. do you know what the reported voltage of 75V is good for? As I read [here](https://github.com/raomin/ESPAltherma/discussions/246#discussioncomment-5360115) and [here](https://github.com/raomin/ESPAltherma#calculating-cop) everybody else seems using that value instead of a fixed 230V value.

**Reply by KalaNagHTD** [reply]
There is definilely a register which reports the right primary voltage of about 230V.
With my HPSU/ECH2O it is {0x21,4,101,2,-1,“Voltage (N-phase) (V)“}.
If you cann‘t find it use the fixed value of 230V.
